backend/internal/system/database/provider/dbprovider.go

```diff
@@ -42,53 +44,113 @@ type dbConfig struct {
 // DBProviderInterface defines the interface for getting database clients.
 type DBProviderInterface interface {
 	GetDBClient(dbName string) (client.DBClientInterface, error)
+	Close() error
 }
 
 // DBProvider is the implementation of DBProviderInterface.
-type DBProvider struct{}
+type DBProvider struct {
+	identityClient client.DBClientInterface
+	runtimeClient  client.DBClientInterface
+	once           sync.Once
+	mu             sync.RWMutex
+}
+
+var (
+	instance *DBProvider
+	once     sync.Once
+)
 
-// NewDBProvider creates a new instance of DBProvider.
-func NewDBProvider() DBProviderInterface {
-	return &DBProvider{}
+// GetDBProvider returns the instance of DBProvider.
+func GetDBProvider() DBProviderInterface {
+	once.Do(func() {
+		instance = &DBProvider{}
+	})
+	return instance
 }
 
 // GetDBClient returns a database client based on the provided database name.
 func (d *DBProvider) GetDBClient(dbName string) (client.DBClientInterface, error) {
```

Comment:
Let's add a comment saying we shouldn't/ generally not needed to close the client